packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2654,30 +2654,25 @@ export class ContainerRuntime
 				return;
 			}
 
-			const messages: {
-				message: InboundSequencedContainerRuntimeMessage;
-				localOpMetadata: unknown;
-			}[] = local
-				? this.pendingStateManager.processPendingLocalBatch(
-						inboundBatch.messages,
-						inboundBatch.batchStartCsn,
-						inboundBatch.emptyBatchSequenceNumber,
-					)
-				: inboundBatch.messages.map((message) => ({ message, localOpMetadata: undefined }));
-			if (messages.length === 0) {
+			// Reach out to PendingStateManager to zip localOpMetadata into the message list if it's a local batch
+			const messagesWithPendingState = this.pendingStateManager.processInboundBatch(
+				inboundBatch,
+				local,
+			);
+			if (messagesWithPendingState.length === 0) {
 				this.ensureNoDataModelChanges(() => this.processEmptyBatch(inboundBatch, local));
-				return;
+			} else {
```

Comment:
It's a pet peeve of mine to put the expected code path inside an else block. It's insignificant so either way is fine

Reply:
Yeah true, maybe I'll flip it in the next PR